packages/fx-core/src/plugins/resource/bot/aadRegistration.ts

```diff
@@ -25,29 +27,34 @@ export class AADRegistration {
 
     const result = new BotAuthCredential();
 
-    // 1. Register a new AAD App.
-    let regResponse = undefined;
-    try {
-      regResponse = await axiosInstance.post(
-        `${AADRegistrationConstants.GRAPH_REST_BASE_URL}/applications`,
-        {
-          displayName: displayName,
-          signInAudience: AADRegistrationConstants.AZURE_AD_MULTIPLE_ORGS,
-        }
-      );
-    } catch (e) {
-      throw new ProvisionError(CommonStrings.AAD_APP, e);
-    }
-
-    if (!regResponse || !utils.isHttpCodeOkOrCreated(regResponse.status)) {
-      throw new ProvisionError(CommonStrings.AAD_APP);
+    if (!objectId && !msAppId) {
+      // 1. Register a new AAD App.
+      let regResponse = undefined;
+      try {
+        regResponse = await axiosInstance.post(
+          `${AADRegistrationConstants.GRAPH_REST_BASE_URL}/applications`,
+          {
+            displayName: displayName,
+            signInAudience: AADRegistrationConstants.AZURE_AD_MULTIPLE_ORGS,
+          }
+        );
+      } catch (e) {
+        throw new ProvisionError(CommonStrings.AAD_APP, e);
+      }
+
+      if (!regResponse || !utils.isHttpCodeOkOrCreated(regResponse.status)) {
+        throw new ProvisionError(CommonStrings.AAD_APP);
+      }
+      result.clientId = regResponse.data.appId;
+      result.objectId = regResponse.data.id;
+    } else {
```

Comment:
NIT: I prefer put the smaller condition block ahead.
```
if (condition) {
    CheckThrowSomethingMissing("objectId", objectId);
    CheckThrowSomethingMissing("msAppId", msAppId);
    result.objectId = objectId;
    result.clientId = msAppId;
    return;
}
```

Reply:
But I think put the unimportant logic in "else" is better

Reply:
Sure, depends on you.

Reply:
It depends on you to make the final decision. What Zhijie suggest is one best practice that return early for non-core logic, so that people can focus on the remaining code, which is all about core business logic. Another benefit of this tip is that it can reduce the number of if/else condition to make the code be more clear